### ISSUE TITLE
Remove Plugin system of CloudComposer

### ISSUE THREAD

**Issue by SunBlack** [PointCloudLibrary/pcl#2721]
I have done a fresh debug build with PCL and always get:

```
Loading  "pcl_cc_tool_euclidean_clustering_debug.dll"
Could not load  "pcl_cc_tool_euclidean_clustering_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_euclidean_clustering_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_fpfh_estimation_debug.dll"
Could not load  "pcl_cc_tool_fpfh_estimation_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_fpfh_estimation_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_normal_estimation_debug.dll"
Could not load  "pcl_cc_tool_normal_estimation_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_normal_estimation_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_organized_segmentation_debug.dll"
Could not load  "pcl_cc_tool_organized_segmentation_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_organized_segmentation_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_sanitize_cloud_debug.dll"
Could not load  "pcl_cc_tool_sanitize_cloud_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_sanitize_cloud_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_statistical_outlier_removal_debug.dll"
Could not load  "pcl_cc_tool_statistical_outlier_removal_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_statistical_outlier_removal_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_supervoxels_debug.dll"
Could not load  "pcl_cc_tool_supervoxels_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_supervoxels_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
Loading  "pcl_cc_tool_voxel_grid_downsample_debug.dll"
Could not load  "pcl_cc_tool_voxel_grid_downsample_debug.dll"
"The plugin 'D:/pcl_build/bin/cloud_composer_plugins/pcl_cc_tool_voxel_grid_downsample_debug.dll' uses incompatible Qt library. (Cannot mix debug and release libraries.)"
```

I checked it already, but all plugins and even CloudComposer are linked against Qt debug libraries, so I have no idea what causes the issue.

But in general: Plugin system of CloudComposer causes a lot of issues (e.g. #2712). At weekend I tried to switch to `AUTOMOC` and got further compile issues with this demo. As far as I see plugins are really short classes, so we could integrate them fixed in CloudComposer. So maybe we don't need a plugin system for this demo. This should solve most problems.

**Comment by taketwo** [comment]
CloudComposer is not maintained by anyone and I don't think that the original contributor is still around. Personally, I'm not familiar with the codebase, so can not say why the plugin system was implemented.

**Comment by taketwo** [comment]
I fine with such a move. @SergioRAgostinho?

**Comment by SergioRAgostinho** [comment]
I'm also fine. I've seen very few question (I actually don't remember any) around here around the CloudComposer, much less specific to the plugins.

**Comment by kunaltyagi** [comment]
Is this a good first issue for someone not familiar with PCL but familiar with C++?

**Comment by larshg** [comment]
I think someone needs a lot of experience with QT as well. Its not that easy to do plugin system in QT. I have looked at it once for a long time ago, but eventually I'll have a look at it.

**Comment by kunaltyagi** [comment]
I think this issue is for removing the plugin in system, not adding one 😄

**Comment by larshg** [comment]
It will also require some effort to incorporate the plugins code into the main app, but it might be more doable, but still require some knowledge about QT :-)

**Comment by larshg** [comment]
Solved by #4682 and #4754.
